drivers/bms_ic/bq769x2/bq769x2_interface.h

```diff
@@ -173,79 +145,78 @@ int bq769x2_subcmd_write_u1(const struct device *dev, const uint16_t subcmd, uin
  * @returns 0 if successful, negative errno otherwise
  */
-int bq769x2_subcmd_write_i2(const struct device *dev, const uint16_t subcmd, int16_t value);
+int bq769x2_datamem_read_u1(const struct device *dev, const uint16_t reg_addr, uint8_t *value);
 
 /**
- * Write 32-bit signed integer via subcommand to bq769x2 IC
+ * Read 16-bit unsigned integer from bq769x2 data memory
  *
- * @param subcmd The subcommand to write the bytes to
- * @param value Value that should be written
+ * @param reg_addr The data memory register address to read the value from
+ * @param value Pointer to where the value should be stored
  *
  * @returns 0 if successful, negative errno otherwise
  */
-int bq769x2_subcmd_write_i4(const struct device *dev, const uint16_t subcmd, int32_t value);
+int bq769x2_datamem_read_u2(const struct device *dev, const uint16_t reg_addr, uint16_t *value);
 
 /**
- * Write 32-bit float via subcommand to bq769x2 IC
+ * Read 8-bit signed integer from bq769x2 data memory
  *
- * @param subcmd The subcommand to write the bytes to
- * @param value Value that should be written
+ * @param reg_addr The data memory register address to read the value from
+ * @param value Pointer to where the value should be stored
  *
  * @returns 0 if successful, negative errno otherwise
  */
-int bq769x2_subcmd_write_f4(const struct device *dev, const uint16_t subcmd, float value);
+static inline int bq769x2_datamem_read_i1(const struct device *dev, const uint16_t reg_addr,
+                                          int8_t *value)
+{
+    return bq769x2_datamem_read_u1(dev, reg_addr, (uint8_t *)value);
+}
 
 /**
- * Read 8-bit unsigned integer from bq769x2 data memory
+ * Read 16-bit signed integer from bq769x2 data memory
  *
  * @param reg_addr The data memory register address to read the value from
  * @param value Pointer to where the value should be stored
  *
  * @returns 0 if successful, negative errno otherwise
  */
-int bq769x2_datamem_read_u1(const struct device *dev, const uint16_t reg_addr, uint8_t *value);
+static inline int bq769x2_datamem_read_i2(const struct device *dev, const uint16_t reg_addr,
+                                          int16_t *value)
+{
+    return bq769x2_datamem_read_u2(dev, reg_addr, (uint16_t *)value);
+}
 
 /**
- * Read 16-bit unsigned integer from bq769x2 data memory
+ * Read 32-bit float from bq769x2 data memory
  *
  * @param reg_addr The data memory register address to read the value from
  * @param value Pointer to where the value should be stored
  *
  * @returns 0 if successful, negative errno otherwise
  */
-int bq769x2_datamem_read_u2(const struct device *dev, const uint16_t reg_addr, uint16_t *value);
+int bq769x2_datamem_read_f4(const struct device *dev, const uint16_t reg_addr, float *value);
 
 /**
  * Write 8-bit unsigned integer to bq769x2 data memory
  *
- * @param reg_addr The data memory register address to write the bytes to
+ * @param reg_addr The data memory register address to
+ * write the bytes to
```

Comment:
Is this formatting intentional, or has something changed with clang-format? (This question applies to the other four occurrences as well.)

Reply:
Yes, clang-format changed it automatically. The behaviour of clang-format is sometimes weird. I wonder why it didn't change the format in the previous PR already, where this code was originally introduced.

I can add it to a dedicated commit if you prefer, but it won't let me keep the old formatting (unless I switch off the auto-formatting in VS code).

Reply:
> Yes, clang-format changed it automatically. The behaviour of clang-format is sometimes weird. I wonder why it didn't change the format in the previous PR already, where this code was originally introduced.


Should we use `clang-format -style=file -dump-config > .clang-format` to dump the entire clang-format file? This could help address issues with updating clang-format, where default values appear to change unexpectedly.

> I can add it to a dedicated commit if you prefer, but it won't let me keep the old formatting (unless I switch off the auto-formatting in VS code).

No need for a separate commit.

Reply:
So you think we should use that dumped `.clang-format` file instead of the existing one because the configuration would be more explicit?

We've experienced in the past that the clang-format result sometimes differs depending on the used version, which is a bit annoying as it means you have to make sure to use exactly the same as in CI.